### ISSUE TITLE
Bad Response: Sequence set is inavlid!

### ISSUE THREAD

**Issue by leopku** [soywod/himalaya#276]
Email provider: https://mail.qq.com/

```
[qq]
default = true
email = <redacted>

imap-host = "imap.qq.com"
imap-port = 993
imap-login = <redacted>
imap-passwd-cmd = <redacted>
# imap-starttls = true
# imap-insecure = true

smtp-host = "smtp.qq.com"
smtp-port = 465
smtp-login = <redacted>
smtp-passwd-cmd = "security find-internet-password -gs gmail -w"
```

```
> himalaya  l
```
Error: cannot fetch messages within range "{}"

Caused by:
    Bad Response: Sequence set is inavlid!

But can read mail by SEQ:

```
> himalaya r 5000
```

**Comment by soywod** [comment]
The `r` shortcut is for `reply`, not `read`. Could you try with `himalaya read 5000`?

**Comment by soywod** [comment]
I discovered that the range is not displayed in this error message. I just fixed this, so with the next release we will be able to see which range your mail provider's IMAP server does not like. I let you know!

**Comment by soywod** [comment]
I also added logs to have more context in debug mode!

**Comment by soywod** [comment]
Could you try again with the `v0.5.2` and share the logs again?

**Comment by leopku** [comment]
Sure. Thank you @soywod .

Error changed this time.

```
Error: cannot fetch messages within range "11609:11590"

Caused by:
    Bad Response: Sequence set is inavlid!
```

**Comment by soywod** [comment]
The range looks good, I don't get why the IMAP server returns such an error. Could you try with `himalaya list -p 1 -s 1`?

**Comment by soywod** [comment]
Also, when you do `himalaya list`, do you see message with id matching the sequence 11609 and 11590?

**Comment by leopku** [comment]
* list with parameter can see message with id and title etc
* list without parameters can NOT see message.

```
❯❯❯ ./himalaya list -p 1 -s 1

ID    │FLAGS │SUBJECT                                                  │SENDER     │DATE
11611 │✷     │<redacted> │<redacted> │2022-02-03 09:15:03

~/Downloads ❯❯❯ ./himalaya list
Error: cannot fetch messages within range "11611:11592"

Caused by:
    Bad Response: Sequence set is inavlid!
```

**Comment by soywod** [comment]
Mmmh is it because of the reverse order of the range? According to the [RFC](https://datatracker.ietf.org/doc/html/rfc3501#section-9) it should not be a problem:

> Example: 2:4 and 4:2 are equivalent and indicate values 2, 3, and 4.

What I can try to do is to create a new branch with the ordered range and see if it works for you. Do you have the rust toolchain installed so you can build from sources?

**Comment by soywod** [comment]
If you really can't, I can publish a minor version with the updated order and see if it changes sth for you!

**Comment by leopku** [comment]
I found only size equal 1 can work.

```
❯❯❯ ~/Downloads/himalaya -a qq list -p 1 -s 1                                                                                      ✘ 1

ID    │FLAGS │SUBJECT                    │SENDER             │DATE
11623 │✷     │<redacted> │<redacted> │2022-02-04 19:34:06
```

Any number larger than 1 can NOT work.

```
❯❯❯ ~/Downloads/himalaya -a qq list -p 1 -s 2                                                                                      ✘ 1
Error: cannot fetch messages within range "11623:11622"

Caused by:
    Bad Response: Sequence set is inavlid!
```

Hope this message is useful.

**Comment by leopku** [comment]
> Mmmh is it because of the reverse order of the range? According to the [RFC](https://datatracker.ietf.org/doc/html/rfc3501#section-9) it should not be a problem:
> 
> > Example: 2:4 and 4:2 are equivalent and indicate values 2, 3, and 4.
> 
> I can try to reorder the range and push the fix on a new branch to see if it works for you. Do you have the rust toolchain installed so you can build from sources?

Unfortunately, I had no experience and toolchain for rust. :(

**Comment by soywod** [comment]
> Unfortunately, I had no experience and toolchain for rust. :(

It's fine don't worry, I will apply the fix and let you know when the next release is ready!

**Comment by soywod** [comment]
I just released the v0.5.4 with the new range order. Let me know if it changes something for you!

**Comment by leopku** [comment]
Big thanks.

It works both with no argument or argument like `-s <int>`
